### PR TITLE
Add comparisons to Elm and PureScript to the FAQ

### PR DESCRIPTION
**Pull request by glennsl**
I don't exactly expect this to go through sans discussion, so have at it! It's also been quite a while since I evaluated them, so there's probably plenty of stuff here that's just plain wrong even.

### CODE REVIEW

File: src/pages/community/faq.md [hunk 1]
```diff
@@ -23,6 +23,12 @@ They'll take your code from the (macOS) clipboard, convert it, and paste it back
...
 
+#### How does Reason (and BuckleScript) compare to Elm
+Reason is a general-purpose language that can target node, native code and utilize both the opam and npm ecossytems. Elm is more opininated and focused, aiming at improving the experience in its own niche and ecosystem, at a rather high cost to its flexibility. Elm has better error messages, though Reason is closing in fast. BuckleScript has superb js interop, generates highly readable and easily debuggable js code and can utilize the npm ecosystem to its full extent, while Elm's FFI and interop is rather convoluted. Elm is pure, while Reason is pragmatic (take that as you wish :)). OCaml has had 25 or so years to mature, and has an active academic base of contributors that keep it close to the forefront of progamming langauge development, while Elm is relatively new and unstable, but moves faster. Reason has first-class support for React, while Elm is focused solely on "The Elm Architecture" (TEA). Reason does however also have a TEA implementation that does a surprisingly good job of mimicing its muse.
```
Comment (OvermindDL1) [inline src/pages/community/faq.md]: > Reason does however also have a TEA implementation that does a surprisingly good job of mimicing its muse.

Yay!  ^.^

Comment (af) [reply src/pages/community/faq.md]: Two suggestions on the Elm section:

* Might as well link to https://github.com/OvermindDL1/bucklescript-tea :)
* "interop is rather convoluted" might be interpreted as combatative– even though I would personally agree with the statement, it might be more diplomatic to use a relative phrase like "is more involved" or "is more complex"

File: src/pages/community/faq.md [hunk 2]
```diff
@@ -23,6 +23,12 @@ They'll take your code from the (macOS) clipboard, convert it, and paste it back
...
+
+#### How does Reason (and BuckleScript) compare to PureScript
+Like Elm, PureScript is relatively new and unstable, more so than Elm. It also targets the JavaScript niche, but unlike Elm it doesn't focus solely on the front-end. Like BuckleScript, PureScript compiles to highly readable JavaScript code, and has significantly better FFI than Elm, though its purity does impose some limitations. PureScript also targets the bower ecosystem instead of npm. Given its name, PureScript is naturally quite heavily invested in its purity, while Reason is and will always be a pragmatic lanaguage. PureScript has several UI frameworks with no clear winner, some of them using or having semantics very close to React, while Reason of course is quite heavily focused on the latter.
```
Comment (jaredly) [inline src/pages/community/faq.md]: - maybe replace "better FFI than Elm" with "easier/less-strict FFI than Elm"
- "the bower ecosystem (which is [not recommended](https://github.com/bower/bower.github.io/commit/12566e5b777ef4c93c525cf3d2fa35ebc9f4d2e5#diff-c35334318579de7b9e052bd7a04ad054R57) by its maintainers)"
- also the pragmatic thing. Dunno if "Given its name, PureScript is naturally quite heavily invested in its purity" adds a ton. maybe "PureScript is pretty much an attempt to bring Haskell to the web (Elm is inspired by Haskell, but diverges in several key areas), and so will be familiar to people coming from Haskell backgrounds. Reason is literally OCaml, but we're putting in a lot of effort to make it accessible to people with little or no functional programming background."

**Comment by chenglou** [comment]
Let's hold this for a little btw. I'd like to make sure we shine a good light on our cousin languages

File: src/pages/community/faq.md (hunk 1 above)
Comment (glennsl) [reply src/pages/community/faq.md]: @jaredly 
* I've tried to integrate your first point with some other points I was making, while being a bit briefer.
* Immutable vs mutable data types is covered by the "pure vs non-pure/pragmatic" point I think, which I realize you'd rather remove, but this is a bit wordy and specific, without addressing the other aspects of purity (which will quickly get you into controversial territory).
* "pragmatic" is admittedly a bit of a weasel word, but Andre completely evacuates the meaning of the words (maybe that was the point though). My intention here is to address what is a very important distinction, acknowledge that the topic is controversial, and at the same time keep it brief. That has to come at some expense to precision. I think being vague is ok as long as we're honest and open about it. It can probably be improved, but I don't think it should just be removed.

File: src/pages/community/faq.md (hunk 2 above)
Comment (glennsl) [reply src/pages/community/faq.md]: * I replaced "easier FFI" with "simpler FFI"
* The bower maintainers seem to primarily address js consumers. And the choice of bower is very deliberate from what I understand: http://harry.garrood.me/blog/purescript-why-bower/
* Your last suggestion is a bit wordy, and goes into the motivations of the PS developers which is a bit of a shaky ground I think, familiarity is an important point I've missed. How about just "Reason, more so than OCaml, aims to be familiar to JavaScript developers, while PureScript is much more flirtatious towards Haskell developers". I'd like to make a similar comparison with Elm, but familiarity doesn't really seem to be a concern for them, even if accessibility very much is, which is a bit contradictory IMO.

File: src/pages/community/faq.md [hunk 3]
```diff
@@ -23,6 +23,21 @@ They'll take your code from the (macOS) clipboard, convert it, and paste it back
...
+* Elm is pure, while Reason is "pragmatic" (take that as you wish :)).
+* OCaml has had 25 or so years to mature, and has an active academic base of contributors that keep it close to the forefront of progamming langauge development, while Elm is relatively new and unstable, but moves faster and has less legacy cruft.
+* Reason has first-class support for React, while Elm is focused solely on "The Elm Architecture" (TEA). Reason does however also have [a TEA implementation](https://github.com/OvermindDL1/bucklescript-tea) that does a surprisingly good job of mimicing its muse.
+
+#### How does Reason (and BuckleScript) compare to PureScript
+* Like Elm, PureScript is relatively new and unstable, more so than Elm. It also targets the JavaScript niche, but unlike Elm it doesn't focus solely on the front-end.
+* Like BuckleScript, PureScript compiles to highly readable JavaScript code, and has a significantly simpler FFI system than Elm, though its purity does impose some limitations.
+* PureScript also targets the bower ecosystem instead of npm.
```
Comment (bassjacob) [inline src/pages/community/faq.md]: this isn't necessarily true anymore, but also isn't as much an issue for purescript. it isn't use bower to replace npm, it just doesn't allow nested resolution of modules so I'm not sure that it matter for the conversation?

Comment (glennsl) [reply src/pages/community/faq.md]: Huh, TIL, do you have a source that expands on that?
